src/js/vic-v2/components/PhotoPreview.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import environment from '../../common/helpers/environment';
+import LoadingIndicator from '../../common/components/LoadingIndicator';
+
+export default class PhotoPreview extends React.Component {
+  constructor(props) {
+    super(props);
+
+    if (props.isLoggedIn && props.id && !props.src) {
+      const userToken = window.sessionStorage.userToken;
+      const headers = {
+        'X-Key-Inflection': 'camel',
+        Authorization: `Token token=${userToken}`
+      };
+
+      fetch(`${environment.API_URL}/v0/vic/profile_photo_attachments/${props.id}`, {
+        headers
+      }).then(resp => {
+        if (resp.ok) {
+          return resp.blob();
+        }
+
+        return new Error(resp.responseText);
+      }).then(blob => {
+        this.props.onUpdatePreview(window.URL.createObjectURL(blob));
+      }).catch(err => {
+        this.props.onError(err);
+      });
+    }
+  }
+
+  render() {
+    const { src, id, className, isLoggedIn, processing } = this.props;
+
+    if (processing) {
+      return (
+        <div className="usa-alert usa-alert-warning vic-processing-warning">
+          <div className="usa-alert-body">
+            <h4 className="usa-alert-title">We're still processing your preview photo</h4>
```

Comment:
My understanding is that we send the photo to our service, that service processes it, and we download the image and use that as the preview.

Reply:
That's what we were doing before, but now the image re-encoding that happens on the backend is done async, so it's possible that we won't be able to get the preview image immediately. Mostly this doesn't matter, because we still have image data in memory, but if you save your form, leave, and come back, it's possible that we'll still be processing.